components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -550,11 +550,18 @@ during post consent handling to get the original ConsentClaimsData object (Assum
                 ConsentClaimsData object.
              */
             ConsentClaimsData value = getConsentRequiredClaims(loggedInUser, serviceProvider, oauth2Params);
+            if (value != null) {
```

Comment:
explain the reason to recalculate the requestedOidcClaimsList in oidc level as a comment.